### PR TITLE
feat: Add Person Preview when clicking a PersonHeader

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Commonly there is a Person ion the page that you are interested in but you don't want to jump straight to them. A little preview would be nice right...?

## Changes

* First click opens a popover with the Person and their properties list
* Second click goes straight to the persons page.
* Modifies Link to respect the `preventDefault()` if the onClick handler triggers it.
* Also randomly fixes PersonHeader to use gravatar if the display name is an email.
* Adds the Person icon to the Person page
* Sorts the properties by default to have custom properties above PostHog ones when sorted

**Persons page**
Screenshot 2023-07-28 at 13 03 35

**Popover**
Screenshot 2023-07-28 at 13 03 28

**Gif**

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## Follow up work

- [ ] Starring of favourite properties to improve ordering even more
- [ ] Improve the reliability of the gravatar lookup by loading the emails of persons _after_ the fact if the properties aren't provided by default

### CODE REVIEW

**Review by annikaschmid — APPROVED** [review]

**Review by pauldambra — APPROVED** [review]
LGTM

For someone who does want to go straight to the person we've gone from one to two clicks... not the worst thing in the world. 

We could have the open person arrow pointing up and to the right button after the person ID so that you can click to open the popover or click to go to the person. But maybe that's a follow up if people complain thing 🤷

File: frontend/__snapshots__/scenes-app-persons-modal--persons-modal.png
Comment (Twixes) [inline frontend/__snapshots__/scenes-app-persons-modal--persons-modal.png:1]: There is some weird wrapping going on here

**Review by Twixes — DISMISSED** [review]

File: frontend/src/scenes/notebooks/Nodes/NodeWrapper.scss [hunk 1]
```diff
@@ -13,10 +13,10 @@
...
     transition: outline 150ms ease-out, outline-offset 150ms ease-out;
 
-    &--selected {
-        outline: 2px solid var(--accent-3000-light);
-        outline-offset: 0.25rem;
-    }
+    // &--selected {
+    //     outline: 2px solid var(--accent-3000-light);
+    //     outline-offset: 0.25rem;
+    // }
```
Comment (Twixes) [inline frontend/src/scenes/notebooks/Nodes/NodeWrapper.scss:19]: Commented out code isn't ideal

File: frontend/src/scenes/persons/PersonDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    content = noPopover ? (
+        content
+    ) : (
+        <Popover
+            overlay={<PersonPreview distinctId={person?.distinct_id || person?.distinct_ids?.[0]} />}
+            visible={visible}
+            onClickOutside={() => setVisible(false)}
+            placement="right"
+            fallbackPlacements={['bottom', 'top']}
+        >
```
Comment (Twixes) [inline frontend/src/scenes/persons/PersonDisplay.tsx:84]: Suggestion: I think the popover would be more clearly connected to the person display if it had `showArrow`

File: frontend/src/scenes/persons/PersonPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { IconOpenInNew } from 'lib/lemon-ui/icons'
+import { asDisplay } from './person-utils'
+
+export type PersonPreviewProps = {
+    distinctId: string | undefined
+}
+
+export function PersonPreview(props: PersonPreviewProps): JSX.Element {
+    if (!props.distinctId) {
+        return <></>
```
Comment (Twixes) [inline frontend/src/scenes/persons/PersonPreview.tsx:18]: Empty fragments are not good for representing empty results, e.g. they're not falsy. `null` is better

File: frontend/src/scenes/persons/PersonPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    const { person, personLoading } = useValues(personLogic({ id: props.distinctId }))
+
+    if (personLoading) {
+        return (
+            <>
+                <Spinner />
+            </>
```
Comment (Twixes) [inline frontend/src/scenes/persons/PersonPreview.tsx:27]: Superfluous fragment

File: frontend/src/lib/lemon-ui/ProfilePicture/ProfilePicture.scss [hunk 5]
```diff
@@ -40,6 +40,13 @@
         font-size: 20px;
     }
 
+    &.lg {
+        width: 32px;
+        height: 32px;
+        line-height: 32px;
+        font-size: 16px;
+    }
```
Comment (Twixes) [inline frontend/src/lib/lemon-ui/ProfilePicture/ProfilePicture.scss:48]: I can see how this is not clear, but the _default_ size already has the same dimensions as this new `lg`, just with a slightly larger font. So now we basically have two "large" sizes, and it's hard to say which to use when. I propose we remove `width`, `height`, `line-height` and `font-size` from the base class, and `.profile-picture`, and enforce there always being a size in `ProfilePicture` with `size = 'lg'`

**Review by Twixes — COMMENTED** [review]
Left some ideas for tweaks.

This wrapping in the persons modal still looks wrong (plus unexpected padding at the bottom):

Screenshot 2023-08-01 at 16 41 54

Subjectively I would also turn down the size of the preview header contents. They look a bit bulky to me, and in particular the profile picture almost touches the edges of the card.

| Current | Maybe smaller? |
| --- | --- |
| big | small | 

(you can see `showArrow` here since I added that on my branch)

File: frontend/src/scenes/notebooks/Nodes/NodeWrapper.scss (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/scenes/notebooks/Nodes/NodeWrapper.scss:19]: This was a bad commit from swapping branches too quick 🙈

**Pull request merged**